server/src/controllers/ScheduleController.js

```diff
@@ -0,0 +1,88 @@
+const ScheduleServices = require('../services/ScheduleServices');
+
+const ScheduleController = {
+  /**
+   * Get an array of schedule objects for a frosh group
+   * @param {Object} req
+   * @param {Object} res
+   * @param {Function} next
+   * @return {Object[]}
+   */
+  async getGroupSchedule(req, res, next) {
+    const { froshGroup } = req.body;
+
+    res.status(200).send({ groupSchedule: await ScheduleServices.getGroupSchedule(froshGroup) });
+  },
+
+  /**
+   * edit a schedule object
+   * @param {Object} req
+   * @param {Object} res
+   * @param {Function} next
+   * @return {Object}
+   */
+  async editSchedule(req, res, next) {
+    const { id, froshGroup, events, date } = req.body;
+    try {
+      res
+        .status(200)
+        .send({ newSchedule: await ScheduleServices.editSchedule(id, froshGroup, events, date) });
+    } catch (e) {
+      next(e);
+    }
+  },
+
+  /**
+   * reorder a schedule object
+   * @param {Object} req
+   * @param {Object} res
+   * @param {Function} next
+   * @return {Promise<void>}
+   */
+  async reorderSchedule(req, res, next) {
+    const { id, order } = req.body;
+
+    try {
+      res.status(200).send({ newSchedule: await ScheduleServices.reorderSchedule(id, order) });
+    } catch (e) {
+      next(e);
+    }
+  },
+
+  /**
+   * add a schedule object
+   * @param {Object} req
+   * @param {Object} res
+   * @param {Function} next
+   * @return {Promise<void>}
+   */
+  async addSchedule(req, res, next) {
```

Comment:
instead of adding a complete schedule, I would rather have an endpoint that allows someone to add an event to the schedule for the frosh group. This endpoint should have some logic that would create a schedule for said frosh group if one does not already exist (https://stackoverflow.com/a/33401897/14216419)